### PR TITLE
Fix #1868: Add Support for rolling update

### LINKED ISSUE #1868: How to do a rolling update
Hi!

What is the fabric8 java client equivalent of:

    kubectl -n=myns rollout restart deploy myapp

I've tried this

    client.apps().deployments().inNamespace(myns).withName("myapp")
				.rolling().updateImage("registry/someimage:0.1");
But get error:

    io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://2c82a54d-405e-44f3-b59c-108b3378232.xxxx.com/apis/apps/v1/namespaces/einnovator/deployments/notifications. Message: Deployment.apps "myapp" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"myapp", "deployment":"851bc111aa0d2f431512f1bfb0fc3f1c"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable. 

Lost here..any help will my valuable...could this be a bug on the fabrix8 java-client implementation?!?

J.

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #1868 #2247

### CODE REVIEW

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/apps/v1/ReplicaSetOperationsImpl.java [hunk 1]
```diff
@@ -78,23 +77,57 @@ public ReplicaSet updateImage(String image) {
...
     if (oldRC.getSpec().getTemplate().getSpec().getContainers().size() > 1) {
       throw new KubernetesClientException("Image update is not supported for multicontainer pods");
     }
-    if (oldRC.getSpec().getTemplate().getSpec().getContainers().size() == 0) {
+    if (oldRC.getSpec().getTemplate().getSpec().getContainers().isEmpty()) {
       throw new KubernetesClientException("Pod has no containers!");
     }
 
-    Container updatedContainer = new ContainerBuilder(oldRC.getSpec().getTemplate().getSpec().getContainers().iterator().next()).withImage(image).build();
+    Container container = oldRC.getSpec().getTemplate().getSpec().getContainers().get(0);
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/apps/v1/ReplicaSetOperationsImpl.java:89]: nit: when retrieving the first item of a collection, in general it's better to do
```suggestion
    Container container = oldRC.getSpec().getTemplate().getSpec().getContainers().iterator().next();
```
This won't be affected by performance issues for specific collection implementations (imagine containers was some sort of LinkedList where access by index was heavily penalized)

**Comment by rohanKanojia** [comment]
retest this please

**Comment by rohanKanojia** [comment]
retest this please

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**